### DISCUSSION TITLE
Es un bug que quede pegado el valor o se debe agregar una línea al question exec?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#193] [2. Questions and Answers] [answered]
Hola en referencia a las discusiones:

https://github.com/Cotalker/documentation/discussions/176
https://github.com/Cotalker/documentation/discussions/186

Salió otra duda, el cliente me informó que en el momento de ingresar los valores para hacer el cálculo, éste funciona sin problemas, pero en el momento que se "borren" los valores de entrada, el resultado queda pegado.

Mi pregunta es:

Eso es un bug visual o se debe agregar algo en el código

Dejo el video de referencia:
**Link:** https://drive.google.com/file/d/1ANJPxgm69FWNGu_3AmSnX8N9TwjN2vGN/view?usp=sharing

Gracias :>

**Comment by acedward** [comment, marked as answer]
@NicolasPLG esto es por el código. 

NOTA: Esto no tiene que ver con Cotalker, sino con la implementación. Podemos ayudarte, pero el código de los Question-Exec es responsabilida del implementador.

Puedes pegar el código aquí y podemos mirarlo.

**Reply by NicolasPLG** [reply]
@acedward sé que es un código, porque no tenía la claridad

Dejo el último code

```
function run() {
  const commands = []
  const peso = Number(context['responses#peso_hepatopancreaticobiliar']);
  console.log('peso:', peso);
  
  const altura = Number(context['responses#talla_hepatopancreaticobiliar']);
  console.log('altura:', altura);
  
  if (!altura || !peso) throw new Error("Missing argument");
  
  var imc = peso / (altura ** 2);
  console.log('imc:', imc);
  
  commands.push({
    cmd: 'SET_RESPONSES',
    value: imc
  });
  commands.push({
    cmd: 'SET_READONLY',
    value: 'true'
  });
  console.log(commands);
  return commands;
}
```

gracias

**Reply by acedward** [reply]
```
function run() {
  const commands = []
  const peso = Number(context['responses#peso_hepatopancreaticobiliar']);
  console.log('peso:', peso);
  
  const altura = Number(context['responses#talla_hepatopancreaticobiliar']);
  console.log('altura:', altura);
  
  if (!altura || !peso) { 
    commands.push({
      cmd: 'SET_RESPONSES',
      value: ''
    });
    commands.push({
      cmd: 'SET_READONLY',
      value: 'true'
    });
    return commands;
  }
  
  var imc = peso / (altura ** 2);
  console.log('imc:', imc);
  
  commands.push({
    cmd: 'SET_RESPONSES',
    value: imc
  });
  commands.push({
    cmd: 'SET_READONLY',
    value: 'true'
  });
  console.log(commands);
  return commands;
}
```

**Reply by acedward** [reply]
Esto podríá funcionar, si no es así dejame saber.

**Reply by NicolasPLG** [reply]
funcionó!!. Gracias @acedward
